### ISSUE TITLE
Expand updating of schedule logic

### ISSUE THREAD

**Issue by mickeelm** [derek-adair/nflgame#18]
Today the update schedule-logic checks the current week, if not explicitly called with arguments/flags to get other weeks.

As per the discussion in #12, we wish to expand this functionality in an attempt to decrease the need for having to run this script manually with flags/arguments. Like this:

* Check which schedules we have in `schedule.json` for the current season.
* If we don't have any schedules for the season, check if PRE0 exists. If not, abort. If PRE0 exists, store PRE0. Check for PRE1. If PRE1 does not exist, abort. If PRE1 exists, store PRE1. Continue like this until all schedules* are fetched (if not aborted).
* If we have _some_ schedules for the season but not all, try to get all missing schedules for the season. Regardless if existing or not, get the current week.
* If we have *all* schedules*, only check the current week.

*) During the period 1st of March to Monday morning (when the late games should have ended) after week 17's games which are all on the Sunday of week 17, _"all schedules"_ means PRE0-PRE4 and REG1-REG17. Not the postseason. The morning after regular season is concluded, _"all schedules"_ includes the postseason.

**Comment by mickeelm** [comment]
@derek-adair you can assign me to this one.

**Comment by mickeelm** [comment]
Right, so after coding this I made some logic changes to make it cleaner programatically, but it still does the same thing as far as I can see.

* Check which weeks we are _missing_ in `schedule.json` for the current season, if any. The missing weeks will be chronologically ordered.
* If the current week is among the missing weeks, add it to the front of the list.
* If the current week is not among the missing weeks, add the current week to the front of the list of missing weeks (which may be empty).
* Try to get the missing weeks, one by one. Abort on failure.*

The above logic should satisfy the initial requirements.

*) This part is a bit special - it will require changes to code that's used in other areas of `nflgame`. As of now, a network failure or an empty schedule (for instance, the URLs for 2019 games exist but they are empty) would both result in a list of empty games being returned to the caller. This caller is `update_week` and looks like this:

```python
def update_week(sched, year, stype, week):
    """
    Updates the schedule for the given week in place. `year` should be
    an integer year, `stype` should be one of the strings `PRE`, `REG`
    or `POST`, and `week` should be an integer in the range `[1, 17]`.
    """
    for game in week_schedule(year, stype, week):
        sched[game['eid']] = game


```
which in turn, I call from `sched.py` in our case (but, there are more callers of this function). The problem is that this function does not return anything, so I can't know if any updates were done or not (it's of course _possible_, but not in an efficient way). This means I have to change the logic in a way to tell the caller (me) that we either got games, or we didn't. If we didn't, abort.

I'm leaning toward using a boolean flag for this, e.g. `false` if no games and `true` if at least one. An exception could be raised too but that might mess things up with other callers. Returning a boolean will not, however. It's up to the caller to decide if it's gonna be used or not.

**Comment by derek-adair** [comment]
Boolean sounds good to me.

I think maybe we should give you a feature branch to work on and commit freely.  I'm unsure if you're wanting me to release this and/or publisht o pypi.

**Comment by mickeelm** [comment]
I'm pretty sure the solution I've made doesn't affect any other functionality than advertised.

Hmm, how do you mean? Do you mean:

a) we should get a branch for you to work on until it's done. _Answer:_ It's done. I have a feature branch called `issue18` on my fork which I've used during development (only one commit on it since I squashed them all) but I am finished. If you merge the PR `nflgame` will work as normal but with this added functionality. At least it does for me. Of course, now that it's redirected to `dev`, some merging needs to be done. Maybe this was a misunderstanding. If you want "ready" PR's to go to `master` directly, then this should have stayed against `master`. I just interpreted it as _never_ create a PR against `master`. Easy to change again, though.

b) I am asking for your permission to publish it. _Answer:_ Please publish it if you feel that it seems to work. It was coded to solve your initial concerns in #12.

**Comment by derek-adair** [comment]
I will merge to dev and poke around a bit.

**Comment by mickeelm** [comment]
Sure, let me know if you run into any problems.

**Comment by derek-adair** [comment]
Closed and released under 1.2.22
